src/dashboard/Data/Browser/Browser.react.js

```diff
@@ -355,6 +356,14 @@ class Browser extends DashboardView {
     }
 
     query.limit(MAX_ROWS_FETCHED);
+    let columns = localStorage.getItem(`ParseDashboard:${VERSION}:${this.props.params.appId}:${source}`)
```

Comment:
I believe we should get it from `ColumnPreferences.getOrder` like in here: https://github.com/parse-community/parse-dashboard/blob/master/src/dashboard/Data/Browser/DataBrowser.react.js#L28

Reply:
I already tried this.
there are missing params at the query point to use it

Reply:
They should be all available at the DataBrowser component. Have you tried to send them up?

Reply:
tried, yes. there is no data anywhere at query point

    console.log('this.refs.dataBrowser', this.refs.dataBrowser);
    console.log('this.state', this.state);
    console.log('this.props', this.props);


<img width="1081" alt="aaa" src="https://user-images.githubusercontent.com/12482438/117034947-7a6b5780-ad0c-11eb-8db9-129de4eda8af.png">

Reply:
tried, yes. there is no data anywhere at query point

    console.log('this.refs.dataBrowser', this.refs.dataBrowser);
    console.log('this.state', this.state);
    console.log('this.props', this.props);

<img width="1081" alt="aaa" src="https://user-images.githubusercontent.com/12482438/117035499-1006e700-ad0d-11eb-9de0-575fa905132c.png">

Reply:
I beleive it should be added in here: https://github.com/parse-community/parse-dashboard/blob/master/src/dashboard/Data/Browser/Browser.react.js#L166

In the same way it is currently doing for the ordering state.

Reply:
if you talk about `getColumnSort` it gives only a single column which class sort by

Reply:
In a similar way you could use the `ColumnPreferences.getOrder` function in that point in order to get the visible columns and set it to the component state.

Reply:
tried and not do the work

    ColumnPreferences.getOrder(
          {},
          this.props.params.appId,
          source
        ))

result `[]`

Reply:
Try with:
```js
   const columnPreferences = context.currentApp.columnPreference || {}
    let order = ColumnPreferences.getOrder(
      {},
      context.currentApp.applicationId,
      props.className,
      columnPreferences[className]
    );
```

Reply:
same

Reply:
it is probably the first parameter. Instead of `{}`, try with `this.getColumns(props.className)`

Reply:
1. I don't find `getColumns` in all project
2. I've try also this:

<img width="468" alt="Screen Shot 2021-05-04 at 8 43 41 PM" src="https://user-images.githubusercontent.com/12482438/117046652-99bcb180-ad19-11eb-88a3-1338bee8ac84.png">


<img width="237" alt="Screen Shot 2021-05-04 at 8 43 51 PM" src="https://user-images.githubusercontent.com/12482438/117046662-a04b2900-ad19-11eb-8172-6d9bbd1cb727.png">

Reply:
Sorry. It is `getClassColumns`

Reply:
its also miss some data:

<img width="876" alt="Screen Shot 2021-05-04 at 8 57 26 PM" src="https://user-images.githubusercontent.com/12482438/117048236-6f6bf380-ad1b-11eb-840a-c316ab25f17e.png">

<img width="515" alt="Screen Shot 2021-05-04 at 8 57 09 PM" src="https://user-images.githubusercontent.com/12482438/117048260-74c93e00-ad1b-11eb-875b-f5d9801696b6.png">

Reply:
It is in the right direction but you will have to figure it out. Probably, when the schema is not fetched yet, we should pass nothing in the first parameter or only consider `columnPreferences[props.className]`. Btw did you try to log the content of this?

Reply:
I'm working on it locally but connected to my production MongoDB
probably, I build the project from my own branch and use it. if you take my solution and fix it - will be community profit

Reply:
ok, I found the right way get columns
code updated